dbms/src/AggregateFunctions/AggregateFunctionUniqCombined.cpp

```diff
@@ -0,0 +1,90 @@
+#include <AggregateFunctions/AggregateFunctionUniqCombined.h>
+
+#include <AggregateFunctions/AggregateFunctionFactory.h>
+#include <AggregateFunctions/Helpers.h>
+
+#include <DataTypes/DataTypeDate.h>
+#include <DataTypes/DataTypeDateTime.h>
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int ILLEGAL_TYPE_OF_ARGUMENT;
+    extern const int NUMBER_OF_ARGUMENTS_DOESNT_MATCH;
+    extern const int ARGUMENT_OUT_OF_BOUND;
+}
+
+namespace
+{
+
+AggregateFunctionPtr createAggregateFunctionUniqCombined(
+    const std::string & name, const DataTypes & argument_types, const Array & params)
+{
+    UInt8 precision = 17; /// default value - must correlate with default ctor of |AggregateFunctionUniqCombinedData|
```

Comment:
If it MUST be correlated it's better to set named constant and use it both here and in ctor.

Reply:
It correlates with a hardcoded union member initialization, the constant won't help - so I left the comment. The code now looks like: `AggregateFunctionUniqCombinedDataWithKey() : set_17() {}` - Also I can use a macro-definition as constant, if it's better.